presto-main/src/main/java/com/facebook/presto/operator/aggregation/arrayagg/GroupArrayAggregationState.java

```diff
@@ -13,165 +13,34 @@
  */
 package com.facebook.presto.operator.aggregation.arrayagg;
 
-import com.facebook.presto.array.IntBigArray;
-import com.facebook.presto.array.ShortBigArray;
-import com.facebook.presto.operator.aggregation.state.AbstractGroupedAccumulatorState;
+import com.facebook.presto.operator.aggregation.AbstractGroupCollectionAggregationState;
+import com.facebook.presto.spi.PageBuilder;
 import com.facebook.presto.spi.block.Block;
-import com.facebook.presto.spi.block.BlockBuilder;
-import com.facebook.presto.spi.block.PageBuilderStatus;
 import com.facebook.presto.spi.type.Type;
-import it.unimi.dsi.fastutil.longs.LongArrayList;
-import it.unimi.dsi.fastutil.longs.LongList;
-import org.openjdk.jol.info.ClassLayout;
+import com.google.common.collect.ImmutableList;
 
-import java.util.ArrayList;
-import java.util.List;
-
-import static com.google.common.base.Verify.verify;
-
-/**
- * state object that uses a single BlockBuilder for all groups.
- */
-public class GroupArrayAggregationState
-        extends AbstractGroupedAccumulatorState
+public final class GroupArrayAggregationState
+        extends AbstractGroupCollectionAggregationState<ArrayAggregationStateConsumer>
         implements ArrayAggregationState
 {
-    private static final int INSTANCE_SIZE = ClassLayout.parseClass(GroupArrayAggregationState.class).instanceSize();
     private static final int MAX_BLOCK_SIZE = 1024 * 1024;
```

Comment:
I was thinking that, because implementations define themselves what the significance of each channel are, this is proprietary knowledge specific to the subclasses and should remain in the subclasses.